[android] NPE 'androidx.lifecycle.MutableLiveData app.organicmaps.widget.placepage.PlacePageViewModel.getMapObject()'

From https://github.com/organicmaps/organicmaps/issues/4549#issuecomment-1437698684

```
02-21 01:18:44.077 31049 31049 I BaseActivityDelegate: MwmActivity: onStart() activity: app.organicmaps.MwmActivity@986726f
02-21 01:18:44.077 31049 31049 I OMcore  : organicmaps/Framework.cpp:863 Java_app_organicmaps_Framework_nativePlacePageActivationListener(): Set global map object listener
02-21 01:18:44.080 31049 31049 I BaseActivityDelegate: MwmActivity: onPostCreate() activity: app.organicmaps.MwmActivity@986726f
02-21 01:18:44.081 31049 31049 I BaseActivityDelegate: MwmActivity: onResume() activity: app.organicmaps.MwmActivity@986726f
02-21 01:18:44.095 31049 31049 I BaseActivityDelegate: MwmActivity: onPostResume() activity: app.organicmaps.MwmActivity@986726f
02-21 01:18:44.118 31049 31049 D AndroidRuntime: Shutting down VM
--------- beginning of crash
02-21 01:18:44.128 31049 31049 E AndroidRuntime: FATAL EXCEPTION: main
02-21 01:18:44.128 31049 31049 E AndroidRuntime: Process: app.organicmaps.beta, PID: 31049
02-21 01:18:44.128 31049 31049 E AndroidRuntime: java.lang.RuntimeException: Unable to destroy activity {app.organicmaps.beta/app.organicmaps.MwmActivity}: java.lang.NullPointerException: Attempt to invoke virtual method 'androidx.lifecycle.MutableLiveData app.organicmaps.widget.placepage.PlacePageViewModel.getMapObject()' on a null object reference
02-21 01:18:44.128 31049 31049 E AndroidRuntime: 	at android.app.ActivityThread.performDestroyActivity(ActivityThread.java:4704)
02-21 01:18:44.128 31049 31049 E AndroidRuntime: 	at android.app.ActivityThread.handleDestroyActivity(ActivityThread.java:4722)
02-21 01:18:44.128 31049 31049 E AndroidRuntime: 	at android.app.servertransaction.DestroyActivityItem.execute(DestroyActivityItem.java:39)
02-21 01:18:44.128 31049 31049 E AndroidRuntime: 	at android.app.servertransaction.TransactionExecutor.executeLifecycleState(TransactionExecutor.java:145)
02-21 01:18:44.128 31049 31049 E AndroidRuntime: 	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:70)
02-21 01:18:44.128 31049 31049 E AndroidRuntime: 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1986)
02-21 01:18:44.128 31049 31049 E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:106)
02-21 01:18:44.128 31049 31049 E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:215)
02-21 01:18:44.128 31049 31049 E AndroidRuntime: 	at android.app.ActivityThread.main(ActivityThread.java:6939)
02-21 01:18:44.128 31049 31049 E AndroidRuntime: 	at java.lang.reflect.Method.invoke(Native Method)
02-21 01:18:44.128 31049 31049 E AndroidRuntime: 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
02-21 01:18:44.128 31049 31049 E AndroidRuntime: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:870)
02-21 01:18:44.128 31049 31049 E AndroidRuntime: Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'androidx.lifecycle.MutableLiveData app.organicmaps.widget.placepage.PlacePageViewModel.getMapObject()' on a null object reference
02-21 01:18:44.128 31049 31049 E AndroidRuntime: 	at app.organicmaps.widget.placepage.PlacePageView.onDestroy(Unknown Source:8)
02-21 01:18:44.128 31049 31049 E AndroidRuntime: 	at androidx.fragment.app.Fragment.performDestroy(Unknown Source:19)
02-21 01:18:44.128 31049 31049 E AndroidRuntime: 	at androidx.fragment.app.FragmentStateManager.destroy(Unknown Source:147)
02-21 01:18:44.128 31049 31049 E AndroidRuntime: 	at androidx.fragment.app.FragmentStateManager.moveToExpectedState(Unknown Source:266)
02-21 01:18:44.128 31049 31049 E AndroidRuntime: 	at androidx.fragment.app.FragmentStore.moveToExpectedState(Unknown Source:30)
02-21 01:18:44.128 31049 31049 E AndroidRuntime: 	at androidx.fragment.app.FragmentManager.moveToState(Unknown Source:27)
02-21 01:18:44.128 31049 31049 E AndroidRuntime: 	at androidx.fragment.app.FragmentManager.dispatchStateChange(Unknown Source:9)
02-21 01:18:44.128 31049 31049 E AndroidRuntime: 	at androidx.fragment.app.FragmentManager.dispatchDestroy(Unknown Source:13)
02-21 01:18:44.128 31049 31049 E AndroidRuntime: 	at androidx.fragment.app.FragmentController.dispatchDestroy(Unknown Source:4)
02-21 01:18:44.128 31049 31049 E AndroidRuntime: 	at androidx.fragment.app.FragmentActivity.onDestroy(Unknown Source:5)
02-21 01:18:44.128 31049 31049 E AndroidRuntime: 	at androidx.appcompat.app.AppCompatActivity.onDestroy(Unknown Source:0)
02-21 01:18:44.128 31049 31049 E AndroidRuntime: 	at app.organicmaps.base.BaseMwmFragmentActivity.onDestroy(Unknown Source:0)
02-21 01:18:44.128 31049 31049 E AndroidRuntime: 	at android.app.Activity.performDestroy(Activity.java:7407)
02-21 01:18:44.128 31049 31049 E AndroidRuntime: 	at android.app.Instrumentation.callActivityOnDestroy(Instrumentation.java:1328)
02-21 01:18:44.128 31049 31049 E AndroidRuntime: 	at android.app.ActivityThread.performDestroyActivity(ActivityThread.java:4689)
02-21 01:18:44.128 31049 31049 E AndroidRuntime: 	... 11 more
```

--- a/android/src/app/organicmaps/widget/placepage/PlacePageView.java
+++ b/android/src/app/organicmaps/widget/placepage/PlacePageView.java
@@ -39,15 +39,12 @@ import app.organicmaps.downloader.CountryItem;
 import app.organicmaps.downloader.DownloaderStatusIcon;
 import app.organicmaps.downloader.MapManager;
 import app.organicmaps.editor.Editor;
-import app.organicmaps.editor.OpeningHours;
-import app.organicmaps.editor.data.Timetable;
 import app.organicmaps.location.LocationHelper;
 import app.organicmaps.location.LocationListener;
 import app.organicmaps.routing.RoutingController;
 import app.organicmaps.settings.RoadType;
 import app.organicmaps.util.SharingUtils;
 import app.organicmaps.util.StringUtils;
-import app.organicmaps.util.ThemeUtils;
 import app.organicmaps.util.UiUtils;
 import app.organicmaps.util.concurrency.UiThread;
 import app.organicmaps.widget.ArrowView;
@@ -138,7 +135,7 @@ public class PlacePageView extends Fragment implements View.OnClickListener,
   };
   private PlacePageViewListener mPlacePageViewListener;
 
-  private PlacePageViewModel viewModel;
+  private PlacePageViewModel mViewModel;
   private MapObject mMapObject;
 
   private static void refreshMetadataOrHide(String metadata, View metaLayout, TextView metaTv)
@@ -167,6 +164,7 @@ public class PlacePageView extends Fragment implements View.OnClickListener,
   @Override
   public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState)
   {
+    mViewModel = new ViewModelProvider(requireActivity()).get(PlacePageViewModel.class);
     return inflater.inflate(R.layout.place_page, container, false);
   }
 
@@ -241,9 +239,8 @@ public class PlacePageView extends Fragment implements View.OnClickListener,
 
     mDownloaderInfo = mPreview.findViewById(R.id.tv__downloader_details);
 
-    viewModel = new ViewModelProvider(requireActivity()).get(PlacePageViewModel.class);
-    viewModel.getMapObject().observe(requireActivity(), this);
-    mMapObject = viewModel.getMapObject().getValue();
+    mViewModel.getMapObject().observe(requireActivity(), this);
+    mMapObject = mViewModel.getMapObject().getValue();
 
     LocationHelper.INSTANCE.addListener(this);
   }
@@ -256,11 +253,11 @@ public class PlacePageView extends Fragment implements View.OnClickListener,
   }
 
   @Override
-  public void onDestroy()
+  public void onDestroyView()
   {
-    super.onDestroy();
+    super.onDestroyView();
     detachCountry();
-    viewModel.getMapObject().removeObserver(this);
+    mViewModel.getMapObject().removeObserver(this);
     LocationHelper.INSTANCE.removeListener(this);
   }
 

